### PR TITLE
Mailet 100 : read only last three commits

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/AddSubjectPrefixTest.java [hunk 1]
```diff
@@ -20,105 +20,66 @@
...
     }
 
-    // test if prefix was added to message without subject
-    public void testSubjectPrefixWasAddedWithoutSubject()
-            throws MessagingException {
-        setupMockedMimeMessage();
-        setupMockedMail(mockedMimeMessage);
-        setupMailet();
+    @Test
+    public void shouldDefinePrefixAsSubjectWhenNoSubject() throws MessagingException {
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/AddSubjectPrefixTest.java]: What about an empty subject?

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java [hunk 2]
```diff
@@ -57,40 +56,8 @@ public void init() throws MessagingException {
...
-                m.setSubject(newSubject); // recover
-            }
-            m.saveChanges();
-        }
-    }
-
-    private String getFirstSubjectHeader(MimeMessage m) throws MessagingException {
-        String delimiter = null;
-        return m.getHeader(RFC2822Headers.SUBJECT, delimiter);
+        m.setSubject(newSubject, Charsets.UTF_8.displayName());
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java]: Not sure to understand. You assume you can always convert to UTF-8? And then you replace the original mail charset by UTF-8?

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/AddSubjectPrefixTest.java (hunk 1 above)
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/AddSubjectPrefixTest.java]: I add the test case

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java (hunk 2 above)
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java]: TL;DR : Yes.
You have to find a charset that can represent the prefix and the original subject. UTF-8 is a easy solution to that problem I guess.

Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/AddSubjectPrefix.java]: Seems fair.

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by mbaechler** [comment]
test this please

**Pull request merged**